lib/spec/staticcacher.sw-spec.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+importScripts('/base/lib/staticcacher.js');
+importScripts('/base/lib/spec/dummies/cachehelper.js');
+
+describe('Static cacher', function () {
+
+  describe('The constructor', function () {
+
+    it('should accept an array of files', function () {
+      var fileList = ['file1', 'file2'];
+      var cacher = new StaticCacher(fileList);
+      expect(cacher.files).to.deep.equal(fileList);
+    });
+
+    it('should throw an error if no argument is provided', function () {
+      expect(function() { new StaticCacher() }).to.throw(Error, 'Invalid file list');
+    });
+
+    it('should throw an error if the argument is not an array', function () {
+      expect(function() { new StaticCacher('invalidString') }).to.throw(Error, 'Invalid file list');
+    });
+
+    it('should throw if the array is empty', function () {
+      expect(function() { new StaticCacher([]) }).to.throw(Error, 'Invalid file list');
```

Comment:
It could be better if we can inform about the specific nature of the error: `File list should contain one item at least`. Wdyt, @arcturus ?